pkg/util/bandwidth/utils.go

```diff
@@ -35,32 +41,69 @@ func validateBandwidthIsReasonable(rsrc *resource.Quantity) error {
 	return nil
 }
 
-// ExtractPodBandwidthResources extracts the ingress and egress from the given pod annotations
-func ExtractPodBandwidthResources(podAnnotations map[string]string) (ingress, egress *resource.Quantity, err error) {
+// bandwidthLimit provides a mapping to cniBandwidthEntry.
+// Rate: The speed knob.
+// Burst: Also known as buffer or maxburst. Size of the bucket.
+// See tc(8) for details.
+type BandwidthLimit struct {
+	Rate, Burst *resource.Quantity
+}
+
+func extractPodBandwidthResources(direction string, podAnnotations map[string]string) (limit *BandwidthLimit, err error) {
 	if podAnnotations == nil {
-		return nil, nil, nil
+		return
 	}
-	str, found := podAnnotations["kubernetes.io/ingress-bandwidth"]
-	if found {
-		ingressValue, err := resource.ParseQuantity(str)
+
+	limit = new(BandwidthLimit)
+	prefix := fmt.Sprintf("kubernetes.io/%s-", direction)
+
+	str, rateFound := podAnnotations[prefix+"bandwidth"]
+	if rateFound {
+		rateValue, err := resource.ParseQuantity(str)
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
-		ingress = &ingressValue
-		if err := validateBandwidthIsReasonable(ingress); err != nil {
-			return nil, nil, err
+		if err := validateBandwidthIsReasonable(&rateValue); err != nil {
+			return nil, err
 		}
+		limit.Rate = &rateValue
 	}
-	str, found = podAnnotations["kubernetes.io/egress-bandwidth"]
-	if found {
-		egressValue, err := resource.ParseQuantity(str)
+
+	str, burstFound := podAnnotations[prefix+"burst"]
+	if burstFound {
+		if !rateFound {
+			return nil, errors.New("rate must be set if burst is set")
+		}
+		burstValue, err := resource.ParseQuantity(str)
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
-		egress = &egressValue
-		if err := validateBandwidthIsReasonable(egress); err != nil {
-			return nil, nil, err
+		if err := validateBandwidthIsReasonable(&burstValue); err != nil {
```

Comment:
does bandwidth validation apply to burst validation?
reading the bandwidth plugin docs 
 https://github.com/containernetworking/plugins/blob/master/plugins/meta/bandwidth/README.md
burst are bits and bandwidth bps

Reply:
Though bandwidth is in bps the units in code are actually both represented as bits since 1 bit per second just equals 1 bit in value, I don't think this differences these two values. 

But I think if we have to do upper bound validation in this step, we have to know what range is rational for a burst value. I think the upper bound of burst shares with bandwidth's, since rationally they all should be the bandwidth of the network interface card. So I think there're no problems sharing this validation.

Actually, I don't think the validation for upper bound is necessary. The current implementation of the bandwidth plugin just feeds this two values to the `tc` command, which doesn't have a limit for these parameters. And the current limit of 1P is fairly large, network interface card bandwidths are just around ten thousand Mbit range after all.

What's your opinion?

Reply:
I personally think we must mimic the `bandwidth` plugin validation and using a name like `validateBandwidthAndBurstIsReasonable`

https://github.com/containernetworking/plugins/blob/cf187287af3f50dfd2de6bea4fec62c1373caf3c/plugins/meta/bandwidth/main.go#L105-L118

Reply:
good catch, thanks.